### PR TITLE
[CARBONDATA-680]Added stats like rows processed in each step. And also fixes unsafe sort enable issue.

### PR DESCRIPTION
**Pull request by ravipesala**
Currently stats like number of rows processed in each step is not added in no kettle flow. This PR adds them.
And also unsafe sort is not enabling even though user enable the sort in property file.This PR fixes it.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/749/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/750/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/751/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/752/

**Comment by jackylk** [comment]
please rebase

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/758/

File: processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java [hunk 1]
```diff
@@ -36,14 +39,37 @@
...
+  private boolean closed;
+
   public AbstractDataLoadProcessorStep(CarbonDataLoadConfiguration configuration,
       AbstractDataLoadProcessorStep child) {
     this.configuration = configuration;
     this.child = child;
+    this.rowCounter = new AtomicLong();
+    this.closed = false;
+
+    new Thread() {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java]: please add comment description for this thread

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/760/

File: processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java (hunk 1 above)
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java]: In this way, even if the info log is not enabled, this thread will be created.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala [hunk 2]
```diff
@@ -484,12 +485,14 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
     val enableReader = {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: suggest change name to vectorizedReader

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala [hunk 3]
```diff
@@ -484,12 +485,14 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
...
+        System.getProperty(CarbonCommonConstants.ENABLE_VECTOR_READER)
       } else {
-        System.getProperty(CarbonCommonConstants.ENABLE_VECTOR_READER,
+        CarbonProperties.getInstance().getProperty(CarbonCommonConstants.ENABLE_VECTOR_READER,
           CarbonCommonConstants.ENABLE_VECTOR_READER_DEFAULT)
       }
     }
     sqlContext.conf.wholeStageEnabled && enableReader.toBoolean &&
-      cols.forall(_.dataType.isInstanceOf[AtomicType])
+    cols.forall(_.dataType.isInstanceOf[AtomicType])
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: This style is not correct, right? It is a condition followed last line

Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: 0k

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala (hunk 2 above)
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/CarbonLateDecodeStrategy.scala]: ok

File: processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java (hunk 1 above)
Comment (ravipesala) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/AbstractDataLoadProcessorStep.java]: Ok, I will check the condition.

**Comment by ravipesala** [comment]
@jackylk Fixed comments, please review

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/763/

**Comment by ravipesala** [comment]
Some problem with single-pass testcases , it hangs the jenkins, I had to manually stop the build.

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/764/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/765/

**Comment by ravipesala** [comment]
@jackylk Fixed comments, please review

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/766/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/770/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/771/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**